### PR TITLE
added sloppy file test

### PR DESCRIPTION
**Pull request by malud**

### CODE REVIEW

File: pkg/converter/sloppy_file_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+var expectedSloppyYml = `project: pkg
+services:
+  apps:
+    busy_env:
+      cmd: sleep 20
+      dependencies:
+      - ../apps/db
+      - ../apps/wordpress
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: I don't think this dependency should be here (only ../apps/db) - sloppy.io dependencies are implemented by Marathon, which are transitive: https://mesosphere.github.io/marathon/docs/rest-api.html#dependencies-array-of-strings

File: pkg/converter/sloppy_file_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      domain: mywords.sloppy.zone
+      env:
+      - WORDPRESS_DB_HOST=db.apps.pkg:3306
+      - WORDPRESS_DB_PASSWORD=wordpress
+      - WORDPRESS_DB_USER=wordpress
+      image: wordpress:4.7.4
+      instances: 1
+      mem: 256
+      port: 80
+      port_mappings:
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: This is redundant to the `port` property. We have it in our json format, but not in the yml format.

File: pkg/converter/sloppy_file_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      mem: 256
+      volumes:
+      - container_path: /var/lib/mysql
+        size: 8GB
+    wordpress:
+      dependencies:
+      - ../apps/db
+      domain: mywords.sloppy.zone
+      env:
+      - WORDPRESS_DB_HOST=db.apps.pkg:3306
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: Where is this mapping implemented? This? https://github.com/sloppyio/sloppose/blob/b272f2360aba544a4152105195fae1dc217c2c94/pkg/converter/linker.go#L71-L77

File: pkg/converter/sloppy_file_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+var expectedSloppyYml = `project: pkg
+services:
+  apps:
+    busy_env:
+      cmd: sleep 20
+      dependencies:
+      - ../apps/db
+      - ../apps/wordpress
+      domain: $URI
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: This line shouldn't be here. The input doesn't have a domain for this app.

File: pkg/converter/testdata/docker-compose-v2.yml [hunk 5]
```diff
@@ -10,8 +10,7 @@ services:
     links:
       - db
   wordpress:
-   depends_on:
-     - db
```
Comment (jzaefferer) [inline pkg/converter/testdata/docker-compose-v2.yml]: Why did you remove this?

File: pkg/converter/sloppy_file_test.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      - MYSQL_DATABASE=wordpress
+      - MYSQL_PASSWORD=wordpress
+      - MYSQL_ROOT_PASSWORD=somewordpress
+      - MYSQL_USER=wordpress
+      image: mysql:8.0.0"
+      instances: 1
+      mem: 256
+      volumes:
+      - container_path: /var/lib/mysql
+        size: 8GB
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: We need a way to specify the volume size, but there is no need to add the default. The size property is optional and the default is filled in on our backend.

File: pkg/converter/sloppy_file_test.go [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      cmd: mysqld
+      domain: $URI
+      env:
+      - MYSQL_DATABASE=wordpress
+      - MYSQL_PASSWORD=wordpress
+      - MYSQL_ROOT_PASSWORD=somewordpress
+      - MYSQL_USER=wordpress
+      image: mysql:8.0.0"
+      instances: 1
+      mem: 256
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: Similar to volume size, instances and mem also have defaults, so there is no need to generate those.

Though it would be good to add memory to one of the services in the input file and match it here.

**Review by jzaefferer — CHANGES_REQUESTED** [review]
Would be great to have a similar test for the v3 docker-compose file format.

File: pkg/converter/sloppy_file_test.go (hunk 2 above)
Comment (malud) [reply pkg/converter/sloppy_file_test.go]: `port_mappings` is there by accident

File: pkg/converter/testdata/docker-compose-v2.yml (hunk 5 above)
Comment (malud) [reply pkg/converter/testdata/docker-compose-v2.yml]: As seen on the busy_box service, services which are declared in `depends_on` and `links` getting also resolved. So to ensure that the dependencies gets set by the host:port search - I've removed this.

File: pkg/converter/sloppy_file_test.go (hunk 3 above)
Comment (malud) [reply pkg/converter/sloppy_file_test.go]: Yes but gets prepared by the buildLinks func (string Template)

File: pkg/converter/sloppy_file_test.go (hunk 1 above)
Comment (malud) [reply pkg/converter/sloppy_file_test.go]: This is a simple test for resolving `depends_on` and `links`

Comment (jzaefferer) [reply pkg/converter/sloppy_file_test.go]: Right, but the input only has `depends_on: db`, not both.

File: pkg/converter/testdata/docker-compose-v2.yml (hunk 5 above)
Comment (jzaefferer) [reply pkg/converter/testdata/docker-compose-v2.yml]: I don't quite understand that part.

Since you mention host:port search: Env vars could reference other hosts (from other services in the same project) without specifying a port. Is that covered anywhere?

File: pkg/converter/sloppy_file_test.go (hunk 1 above)
Comment (malud) [reply pkg/converter/sloppy_file_test.go]: and `links` ? there is wordpress declared. should it be ignored completely ?

Comment (jzaefferer) [reply pkg/converter/sloppy_file_test.go]: Wasn't `links` deprecated anyway? If its safe to simply ignore it, I'd be fine with that.

File: pkg/converter/sloppy_file_test.go [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      dependencies:
+      - ../apps/db
+      domain: mywords.sloppy.zone
+      env:
+      - WORDPRESS_DB_HOST=db.apps.sloppy-test:3306
+      - WORDPRESS_DB_PASSWORD=wordpress
+      - WORDPRESS_DB_USER=wordpress
+      image: wordpress:4.7.4
+      port: 80
+version: v1
```
Comment (jzaefferer) [inline pkg/converter/sloppy_file_test.go]: This should be the first line.

File: pkg/converter/testdata/docker-compose-v3-simple.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+     - "8000:80"
+   restart: always
+   environment:
+     WORDPRESS_DB_HOST: db:3306
+     WORDPRESS_DB_USER: wordpress
+     WORDPRESS_DB_PASSWORD: wordpress
+   entrypoint: ["/code/entrypoint.sh", "-foo", "bar"]
+   deploy:
+     mode: replicated
+     replicas: 3
```
Comment (jzaefferer) [inline pkg/converter/testdata/docker-compose-v3-simple.yml:25]: Shouldn't this and the memory properties below get mapped to `instances` and `mem`?

May need a 2nd test case (non-simple) to be able to compare this to the v2 input.

File: pkg/converter/linker_test.go [hunk 10]
```diff
@@ -9,76 +9,70 @@ import (
...
+func TestLinker_FindService(t *testing.T) {
+	cases := map[string]struct {
+		value       string
+		shouldMatch bool
+	}{
+		"FOO":      {"bar", false},
+		"BAR":      {"foo:80", true},
+		"SHORT":    {"s:80", true},
+		"FOO_BAR":  {"another.foo:443", true},
+		"FOO_HOST": {"bar", true},
```
Comment (jzaefferer) [inline pkg/converter/linker_test.go:21]: Requiring the string 'HOST' in the env var key is a pretty harsh limitation to. Took me a while to understand that based on the implementation. Need to at least add this to the readme, maybe a "Limitations" section.

**Review by jzaefferer — CHANGES_REQUESTED** [review]
Mostly reviewing tests...

File: pkg/converter/testdata/docker-compose-v3-simple.yml (hunk 9 above)
Comment (malud) [reply pkg/converter/testdata/docker-compose-v3-simple.yml:25]: Yes, but since there is no decision handling this from v2 it gets currently ignored. Can be refined later on.

File: pkg/converter/linker_test.go (hunk 10 above)
Comment (malud) [reply pkg/converter/linker_test.go:21]: I have created an issue for this: https://github.com/sloppyio/sloppose/issues/2

File: pkg/converter/sloppy_file_test.go (hunk 8 above)
Comment (malud) [reply pkg/converter/sloppy_file_test.go]: Humanised yes, but its still generated output and parseable from sloppy/cli

File: pkg/converter/testdata/docker-compose-v3-simple.yml (hunk 9 above)
Comment (jzaefferer) [reply pkg/converter/testdata/docker-compose-v3-simple.yml:25]: Could put that into a separate issue as well.

File: pkg/converter/sloppy_file_test.go (hunk 8 above)
Comment (jzaefferer) [reply pkg/converter/sloppy_file_test.go]: Okay, but let's fix that anyway.

File: pkg/converter/testdata/docker-compose-v3-simple.yml (hunk 9 above)
Comment (malud) [reply pkg/converter/testdata/docker-compose-v3-simple.yml:25]: Resource implementation: https://github.com/sloppyio/sloppose/issues/3

**Review by jzaefferer — APPROVED** [review]
